### PR TITLE
docs: split atom creators in multiple files

### LINKED ISSUE #2082: Split atom creators recipe in docs into multiple markdown files
It's a single file for now and it's fairly big.

### PR DESCRIPTION
**Pull request by gerzonc**
## Related Issues or Discussions

Closes #2082

## Summary

Atom creators file in docs is too big so I just split it in a file per atom creator.

## Check List

- [x] `yarn run prettier` for formatting code and docs

### CODE REVIEW

**Comment by gerzonc** [comment]
@dai-shi the other way I thought of doing it is that we can keep it as it was (just atom-creators.mdx) and import all the creators as components, e.g.

```
---
title: Atom creators
nav: 8.02
keywords: creators
---

<AtomWithToggle />
```

I did it this way first, but let me know which one you prefer since I already had both solutions in mind 🙈

**Comment by dai-shi** [comment]
I think multiple files are better for search.

Maybe we should consider about the sidebar labels.
image

@sandren Do you have any idea?

File: docs/recipes/atom-with-storage.mdx [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  return derivedAtom
+}
+```
+
+And how it's used:
+
+```js
+import { atomWithToggleAndStorage } from 'XXX'
+
+// will have an initial value set to false & get stored in localStorage under the key "isActive"
```
Comment (himself65) [inline docs/recipes/atom-with-storage.mdx:40]: set to false? I don't see the initial value on your code

File: docs/recipes/atom-with-listeners.mdx [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+import { atom, useSetAtom, Getter, Setter, SetStateAction } from 'jotai'
+
+type Callback<Value> = (
+  get: Getter,
+  set: Setter,
+  newVal: Value,
+  prevVal: Value
+) => void
+
+export function atomWithListeners<Value>(initialValue: Value) {
```
Comment (himself65) [inline docs/recipes/atom-with-listeners.mdx:29]: I think `useCurrentStore` (non-exist API) and `store.sub(aAtom)` would be better

Comment (dai-shi) [reply docs/recipes/atom-with-listeners.mdx:29]: This recipe is written for v1 API, I guess. `useStore` exists, btw.

File: docs/recipes/atom-with-storage.mdx (hunk 1 above)
Comment (dai-shi) [reply docs/recipes/atom-with-storage.mdx:40]: I think L19 should be `initialValue = false,`.

**Review by dai-shi — COMMENTED** [review]
This PR is to split markdown files. We should probably open a new PR for improving the contents.

**Comment by gerzonc** [comment]
> This PR is to split markdown files. We should probably open a new PR for improving the contents.

I can do both if you don't mind. I'm still waiting for approval of current changes so I can open a new PR in order to update the docs

**Comment by sandren** [comment]
> @sandren Do you have any idea?

Yes, let's change the frontmatter title of each of the new docs to reflect the recipe function name instead. For example, `atomWithToggle` instead of  `Atom creator with toggle`. Once that is done, we can merge. Thanks, @gerzonc! 🙂

**Comment by sandren** [comment]
> Is ready now!

Merged. Thank you so much! 🙂

File: docs/recipes/atom-creators.mdx [hunk 3]
```diff
@@ -1,521 +0,0 @@
...
-      {/* Clicking this button twice will only trigger one render */}
-      <button onClick={() => setStyles({ ...styles, fontSize: 32 })}>
-        Enlarge font
-      </button>
-    </div>
-  )
-}
-```
-
-## atomWithRefresh
```
Comment (dai-shi) [inline docs/recipes/atom-creators.mdx:160]: @gerzonc It seems like this section is lost. Can you double check please?

Comment (gerzonc) [reply docs/recipes/atom-creators.mdx:160]: True OMG 🥺 I'll add it right away!

**Pull request merged**